scipy/signal/tests/test_spectral.py

```diff
@@ -781,6 +781,36 @@ def test_padded_freqs(self):
         assert_allclose(f, fodd)
         assert_allclose(f, feven)
 
+
+class TestTfestimate:
+    def test_identical_input(self):
+        fs = 10e3
+        N = 1e5
+        time = np.arange(N) / fs
+        x = np.sin(2 * np.pi * time)
+        f, Tf = signal.tfestimate(x, x, fs=fs)
+        assert_allclose(Tf, 1)
+
+    def test_exactmatch(self):
+        # Filter random noise x with a Butterworth filter
+        fs = 10e3
+        N = 1e6
+        noise_power = 0.001 * fs / 2
+        time = np.arange(N) / fs
+        x = np.random.normal(scale=np.sqrt(noise_power), size=time.shape)
+        b, a = signal.butter(2, 0.5, 'low')
+        y = signal.lfilter(b, a, x)
+
+        # Estimate the filter from x and y containing same number of samples
+        f, Tf = signal.tfestimate(x, y, fs=fs, nperseg=len(x))
+
+        # Compute the expected finite impulse response of the filter
+        delta = np.zeros(time.shape)
+        delta[0] = 1.0
+        Tf_act = np.fft.rfft(signal.lfilter(b, a, delta))
+        assert_allclose(np.abs(Tf), np.abs(Tf_act), atol=0.01)
```

Comment:
Okay thanks. Just noting that both `TestCoherence` and `TestSpectrogram` from this same file use random numbers without fixing the seed. Would you like me to address that in this PR, or leave it for somebody later?